### PR TITLE
Report manager new ui

### PR DESCRIPTION
**Pull request by waiterZen**
new report ui

### CODE REVIEW

File: frontend/express/public/stylesheets/main.css [hunk 1]
```diff
@@ -1947,6 +1947,7 @@ input.custom-styled-green-check:checked {border-color: #C2C2C2;}
 .date-picker-component .date-picker .calendar-container {
     padding: 14px;
 }
+<<<<<<< Updated upstream
```
Comment (ar2rsawseen) [inline frontend/express/public/stylesheets/main.css]: This looks like an unresolved merge

File: api/utils/taskmanager.js [hunk 2]
```diff
@@ -228,6 +228,18 @@ var request = require("request");
...
+    * @param {object} options - options for the task
+    * @param {object} options.db - database connection
+    * @param {object} options.task_id - ID of the target task 
+    * @param {string} options.data - data of the task want to modify
+    */
+    taskmanager.editTask = function(options, callback){
+        options.db = options.db || common.db;
+        options.db.collection("long_tasks").update({_id: options.task_id}, {$set: options.data}, callback);
+    };
+
```
Comment (ar2rsawseen) [inline api/utils/taskmanager.js]: Let's keep parameter naming consistent with other methods, like `saveResult`, so not task_id, but id, etc

File: frontend/express/public/javascripts/countly/countly.views.js [hunk 3]
```diff
@@ -3775,183 +3775,403 @@ window.DownloadView = countlyView.extend({
...
+            this.renderTable();
+            this.initDrawer();
+        }
+        if (self.taskCreatedBy === 'manually') {
+            $("#report-manager-graph-description").text(jQuery.i18n.map['taskmanager.manually-table-remind'])
+        } else {
+            $("#report-manager-graph-description").text(jQuery.i18n.map['taskmanager.automatically-table-remind'])
+        }
+        const manuallyColumns       = [true, true, true, false, true, true,  true, true, true,  false,  false]
+        const automaticallyColumns  = [false,false,true, true,  true, false, true, false, false, true,  true]
```
Comment (ar2rsawseen) [inline frontend/express/public/javascripts/countly/countly.views.js]: No const in browser

File: frontend/express/public/stylesheets/main.css [hunk 4]
```diff
@@ -2110,4 +2110,17 @@ input.custom-styled-green-check:checked {border-color: #C2C2C2;}
...
+#report-manager-view .report-tab{padding:0;}
+#report-manager-view .ui-tabs .ui-tabs-panel{padding:0;}
+#report-manager-view .ui-tabs .ui-tabs-nav li a{background-color: transparent;  color: #666; padding: 0px 0px 6px 0px;  font-size:15px}
+#report-manager-view .ui-tabs.ui-widget-content .ui-tabs-nav > .ui-state-default {border-top: none;}
+#report-manager-view .ui-tabs .ui-tabs-nav li.ui-tabs-selected {border-top: none; border-bottom: none !important; background-color:transparent;}
+#report-manager-view .ui-tabs .ui-tabs-nav li.ui-tabs-selected a{color:#333; border-bottom: 2px solid #19cc63 !important;}
+#report-manager-view .ui-tabs .ui-tabs-nav li {border-bottom:none !important; width: auto; margin-right: 30px;}
+#report-manager-view .ui-tabs .ui-tabs-nav {margin-top:0px; margin-top:10px; margin-bottom: 10px}
+#report-manager-view #d-table-funnels_wrapper {margin:0px;}
+#report-manager-view #d-table-funnels_wrapper .dataTable-top {border-radius:0px;}
```
Comment (ar2rsawseen) [inline frontend/express/public/stylesheets/main.css]: Why funnels?

File: frontend/express/public/javascripts/countly/countly.views.js [hunk 5]
```diff
@@ -3782,183 +3782,403 @@ window.DownloadView = countlyView.extend({
...
+        if (self.taskCreatedBy === 'manually') {
+            $("#report-manager-graph-description").text(jQuery.i18n.map['taskmanager.manually-table-remind'])
+        } else {
+            $("#report-manager-graph-description").text(jQuery.i18n.map['taskmanager.automatically-table-remind'])
+        }
+        var manuallyColumns       = [true, true, true, false, true, true,  true, true, true,  false,  false]
+        var automaticallyColumns  = [false,false,true, true,  true, false, true, false, false, true,  true]
+        
+        if (self.taskCreatedBy === 'manually') {
+            manuallyColumns.forEach((vis,index)=>{
```
Comment (ar2rsawseen) [inline frontend/express/public/javascripts/countly/countly.views.js]: No arrow functions in browser

File: frontend/express/public/javascripts/countly/countly.views.js [hunk 6]
```diff
@@ -3782,183 +3782,403 @@ window.DownloadView = countlyView.extend({
...
+                this.dtable.fnSetColumnVis( index, vis );
+            })
+            $(".report-manager-idget-header .filter2-segmentation").show()
+            $(".report-manager-idget-header .filter3-segmentation").hide()
+        } else {
+            automaticallyColumns.forEach((vis,index)=>{
```
Comment (ar2rsawseen) [inline frontend/express/public/javascripts/countly/countly.views.js]: Same here

File: frontend/express/public/javascripts/countly/countly.views.js [hunk 7]
```diff
@@ -3782,183 +3782,403 @@ window.DownloadView = countlyView.extend({
...
+        this.states = {
             "all": jQuery.i18n.map["common.all"],
             "running":jQuery.i18n.map["taskmanager.running"],
             "rerunning":jQuery.i18n.map["taskmanager.rerunning"],
             "completed":jQuery.i18n.map["taskmanager.completed"],
             "errored":jQuery.i18n.map["taskmanager.errored"]
         };
+    },
+    beforeRender: function() {
+        return $.when(countlyTaskManager.initialize(null, {"report_desc": {$ne:""}})).then(function () {});
```
Comment (ar2rsawseen) [inline frontend/express/public/javascripts/countly/countly.views.js]: Do we intend to initialize task data only with tasks that have descriptions?

File: frontend/express/public/javascripts/countly/countly.views.js [hunk 8]
```diff
@@ -3782,183 +3782,403 @@ window.DownloadView = countlyView.extend({
...
+
+        $("#create-report").off("click").on("click", function(){
+            var report_id =  $("#current_report_id").text();
+            var canSubmit = self.reporInputValidator();
+            if(!canSubmit){
+                return;
+            }
+            var report_name = $("#report-name-input").val();
+            var report_desc = $("#report-desc-input").val();
+            var global = $("#report-global-option").hasClass("selected");
```
Comment (ar2rsawseen) [inline frontend/express/public/javascripts/countly/countly.views.js]: in theory global is not a reserved word, but very close to it, as could become in future, etc. 
Only for variable can we use something else? Like globalAccess.
It is ok for .global properties, no need to change there

Comment (waiterZen) [reply frontend/express/public/javascripts/countly/countly.views.js]: Updated

File: frontend/express/public/javascripts/countly/countly.views.js (hunk 7 above)
Comment (waiterZen) [reply frontend/express/public/javascripts/countly/countly.views.js]: Now with descriptions are recognized as  manual created

File: frontend/express/public/javascripts/countly/countly.views.js (hunk 6 above)
Comment (waiterZen) [reply frontend/express/public/javascripts/countly/countly.views.js]: updated

File: frontend/express/public/javascripts/countly/countly.views.js (hunk 5 above)
Comment (waiterZen) [reply frontend/express/public/javascripts/countly/countly.views.js]: updated

File: frontend/express/public/javascripts/countly/countly.views.js [hunk 9]
```diff
@@ -3782,183 +3782,403 @@ window.DownloadView = countlyView.extend({
...
+            return false;
+        }
+        $("#create-report").removeClass("disabled")
+        return true;
+    },
+    loadReportDrawerView: function(id){
+        $("#current_report_id").text(id);
+        var data = countlyTaskManager.getResults();
+        for (var i = 0; i < data.length; i++) {
+            console.log(data[i]);
```
[the remark this answers is not in the document]
Comment (waiterZen) [reply frontend/express/public/javascripts/countly/countly.views.js]: updated

File: api/utils/requestProcessor.js [hunk 10]
```diff
@@ -562,7 +562,7 @@ const processRequest = (params) => {
                                 taskmanager.editTask({
                                     db: common.db,
                                     data: data,
-                                    id: params.qstring.id
+                                    task_id: params.qstring.task_id
```
[the remark this answers is not in the document]
Comment (waiterZen) [reply api/utils/requestProcessor.js]: updated

File: frontend/express/public/javascripts/countly/countly.views.js (hunk 7 above)
Comment (ar2rsawseen) [reply frontend/express/public/javascripts/countly/countly.views.js]: So per this code, on view render, in the table we will list only tasks where description is not empty. why?

**Pull request merged**